src/main/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptor.java

```diff
@@ -205,26 +210,28 @@ public ListBoxModel doFillHardwareIdItems(
             @QueryParameter String hardwareId,
             @QueryParameter String endPointUrl, @QueryParameter String identity, @QueryParameter String credential, @QueryParameter String zone
     ) {
-
         ListBoxModel m = new ListBoxModel();
         m.add("None specified", "");
-
+        final String valueOrEmpty = Util.fixNull(hardwareId);
+        boolean existingValueFound = valueOrEmpty.isEmpty();
         try {
-            final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
-            for (Flavor flavor : openstack.getSortedFlavors()) {
-                m.add(String.format("%s (%s)", flavor.getName(), flavor.getId()), flavor.getId());
+            if (haveServerDetails(endPointUrl, identity, credential, zone)) {
+                final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
+                for (Flavor flavor : openstack.getSortedFlavors()) {
+                    final String value = flavor.getId();
+                    final String displayText = String.format("%s (%s)", flavor.getName(), value);
+                    m.add(displayText, value);
+                    existingValueFound |= valueOrEmpty.equals(value);
+                }
             }
-            return m;
         } catch (AuthenticationException | FormValidation | ConnectionException ex) {
             LOGGER.log(Level.FINEST, "Openstack call failed", ex);
         } catch (Exception ex) {
             LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
         }
-
-        if (Util.fixEmpty(hardwareId) != null) {
+        if (!existingValueFound) {
```

Comment:
The intent is that the existing value will always appear as a selectable option regardless of our ability to communicate with OpenStack (as that might be offline).
If we can communicate with OpenStack then we may be able to display the configured value in a better format, e.g. String.format("%s (%s)", flavor.getName(), value), but we will always give the user the option of keeping the configuration unchanged.